src/sys/options.cxx

```diff
@@ -60,8 +60,21 @@ void Options::set(const string &key, const string &val, const string &source) {
   opt.value = val;
   opt.source = source;
   opt.used = false;
-
-  options[lowercase(key)] = opt;
+  string key_l = lowercase(key);
+  if (this->isSet(key_l)) {
+    const OptionValue & oldopt=options[key_l];
+    if (oldopt.value != opt.value) {
+      if (oldopt.source == opt.source) {
+        throw BoutException("Options: Setting a value from same source (%s) to new value %s - old value was %s.",source.c_str(),oldopt.value.c_str(),opt.value.c_str());
+      } else {
+        output_warn << "\tOption " << sectionName << ":" << key << " = " << oldopt.value
+                    << " (" << oldopt.source << ") overwritten with:" << endl
```

Comment:
Not a problem here, but just to note that `std::endl` also flushes to stdout immediately. `\n` is fine (and usually better) to use for a newline.